drivers/flash/Kconfig

```diff
@@ -18,7 +18,7 @@ menuconfig FLASH
 
 config FLASH_PAGE_LAYOUT
 	bool "API for retrieving the layout of pages"
-	depends on FLASH
+	depends on FLASH && (SOC_FLASH_NRF5 || SOC_FLASH_STM32)
```

Comment:
We should do as a HAS_FLASH_PAGE_LAYOUT, don't want to add every SOC here

Reply:
@galak can you elaborate? perhaps point me to an example and I can fix it accordingly

Reply:
Look at how we do CPU_HAS_MPU